### PR TITLE
Fixes #16757: Update the rudder-setup upgrade to support 6.0 upgrade

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/16757

### CODE REVIEW

File: scripts/rudder-setup/setup-server.sh [hunk 1]
```diff
@@ -68,7 +68,7 @@ upgrade_server() {
   # Upgrade
   if [ "${PM}" = "apt" ]
   then
-    ${PM_UPGRADE} rudder-server-root ncf ncf-api-virtualenv
+    ${PM_UPGRADE} rudder-server-root
   else
     ${PM_UPGRADE} "rudder-*" "ncf*"
```
Comment (peckpeck) [inline scripts/rudder-setup/setup-server.sh]: not ncf on 6.0

File: scripts/rudder-setup/setup-server.sh [hunk 2]
```diff
@@ -68,7 +68,7 @@ upgrade_server() {
   # Upgrade
   if [ "${PM}" = "apt" ]
   then
-    ${PM_UPGRADE} rudder-server-root ncf ncf-api-virtualenv
+    ${PM_UPGRADE} rudder-server-root
```
Comment (peckpeck) [inline scripts/rudder-setup/setup-server.sh]: debian needs a dist-upgrade on 6.0

Comment (Fdall) [reply scripts/rudder-setup/setup-server.sh]: a apt install is enough on 6.0

**Comment by Fdall** [comment]
PR updated with a new commit

File: scripts/rudder-setup/setup-server.sh [hunk 3]
```diff
@@ -66,12 +66,7 @@ upgrade_server() {
...
   # Upgrade
-  if [ "${PM}" = "apt" ]
-  then
-    ${PM_UPGRADE} rudder-server-root ncf ncf-api-virtualenv
-  else
-    ${PM_UPGRADE} "rudder-*" "ncf*"
-  fi
+  ${PM_UPGRADE} rudder-server-root
 
   service_cmd rudder-jetty restart
```
Comment (peckpeck) [inline scripts/rudder-setup/setup-server.sh]: why do we need a restart here ?

Comment (Fdall) [reply scripts/rudder-setup/setup-server.sh]: No idea, should not be needed and may hide some upgrade issues

**Comment by Fdall** [comment]
PR updated with a new commit

**Pull request merged**